public_html/wp-content/plugins/wordcamp-speaker-feedback/includes/cron.php

```diff
@@ -0,0 +1,203 @@
+<?php
+
+namespace WordCamp\SpeakerFeedback\Cron;
+
+use function WordCamp\SpeakerFeedback\Comment\{ get_feedback, update_feedback };
+use function WordCamp\SpeakerFeedback\Post\{ get_session_speaker_user_ids, get_session_feedback_url };
+
+defined( 'WPINC' ) || die();
+
+add_action( 'init', __NAMESPACE__ . '\schedule_jobs' );
+add_action( 'sft_notify_speakers_approved_feedback', __NAMESPACE__ . '\notify_speakers_approved_feedback' );
+
+/**
+ * Add cron jobs to the schedule.
+ *
+ * @return void
+ */
+function schedule_jobs() {
+	if ( ! wp_next_scheduled( 'sft_notify_speakers_approved_feedback' ) ) {
+		$next_time = strtotime( 'Next day 6pm ' . wp_timezone_string() );
+		wp_schedule_single_event( $next_time, 'sft_notify_speakers_approved_feedback' );
+	}
+}
+
+/**
+ * Notify speakers via email about newly approved feedback submissions.
+ *
+ * @return void
+ */
+function notify_speakers_approved_feedback() {
+	if ( ! speaker_notifications_are_enabled() ) {
+		return;
+	}
+
+	$wordcamp_name = get_wordcamp_name( get_current_blog_id() );
+
+	$feedbacks       = get_unnotified_feedback();
+	$post_ids        = array_combine(
+		// Using the 'index_key' parameter doesn't work here, maybe because the values are numeric?
+		wp_list_pluck( $feedbacks, 'comment_ID' ),
+		wp_list_pluck( $feedbacks, 'comment_post_ID' )
+	);
+	$unique_post_ids = array_unique( array_values( $post_ids ) );
+
+	$speaker_session_ids = array_reduce(
+		$unique_post_ids,
+		function( $carry, $post_id ) {
+			$user_ids = get_session_speaker_user_ids( $post_id );
+
+			foreach ( $user_ids as $user_id ) {
+				if ( ! isset( $carry[ $user_id ] ) ) {
+					$carry[ $user_id ] = array();
+				}
+				$carry[ $user_id ][] = $post_id;
+			}
+
+			return $carry;
+		},
+		array()
+	);
+
+	foreach ( $speaker_session_ids as $user_id => $session_ids ) {
+		$speaker = get_user_by( 'id', $user_id );
+		// Todo: Implement a UI for toggling this user meta value.
+		if ( true === $speaker->sft_speaker_notifications_opt_out ) {
```

Comment:
Tried testing this with `wp user meta set 4 sft_speaker_notifications_opt_out true`, and it set the value as a string. I'm not sure if setting meta directly will preserve type, but you might want to switch this to `wp_validate_boolean( … )`

Reply:
> This will opt the user out for all WordCamps in the future too, right?

Yeah, I believe this will be a value stored in the usermeta table that is shared across the entire WordPress/WordCamp network. So if the speaker opts out of notifications on one site, they will be opted out on other future WordCamp sites as well. But they'd also be able to opt back in from any site.

> you might want to switch this to wp_validate_boolean( … )

Good idea. ecbd74fd